packages/stratimikos/adapters/belos/src/BelosThyraAdapter.hpp

```diff
@@ -290,17 +290,23 @@ namespace Belos {
 
       const int m = B.numRows();
       const int n = B.numCols();
-      auto vs = A.domain();
-      // Create a view of the B object!
-      Teuchos::RCP< const TMVB >
-        B_thyra = vs->createCachedMembersView(
-          RTOpPack::ConstSubMultiVectorView<ScalarType>(
-            0, m, 0, n,
-            arcpFromArrayView(arrayView(&B(0,0), B.stride()*B.numCols())), B.stride()
-            )
-          );
-      // perform the operation via A: mv <- alpha*A*B_thyra + beta*mv
-      Thyra::apply<ScalarType>(A, Thyra::NOTRANS, *B_thyra, Teuchos::outArg(mv), alpha, beta);
+      // Check if B is 1-by-1, in which case we can just call MvAddMv()
+      if ((m == 1) && (n == 1)) {
+        const ScalarType alphaNew = alpha * B(0, 0);
+        MvAddMv(alphaNew, A, beta, mv, mv);
```

Comment:
@cgcgcg Why not just replace this line with `Thyra::linear_combination<ScalarType>` from the `MvAddMv` method directly?

Reply:
If that's your preference I can definitely do that!

Reply:
@cgcgcg It seems cleaner to me and removes a recursion of timers.  If you don't mind.  :)